src/nvim/charset.c

```diff
@@ -1740,11 +1759,8 @@ char_u* skiptowhite_esc(char_u *p) {
 /// @return Number read from the string.
 intmax_t getdigits(char_u **pp)
 {
-  errno = 0;
-  intmax_t number = strtoimax((char *)*pp, (char **)pp, 10);
-  if (number == INTMAX_MAX || number == INTMAX_MIN) {
```

Comment:
The check for `INTMAX_MAX`/`INTMAX_MIN` is a little redundant if we reset `errno` before. On the other hand, a strict reading of `strtoimax(3p)` would theoretically allow an implementation to set `errno` to `ERANGE` even if an overflow *didn't* occur (returning a value other than the extremal ones). I'll re-add the check.